src/vivarium_public_health/metrics/mortality.py

```diff
@@ -167,23 +149,50 @@ def calculate_ylls(self, x: pd.DataFrame) -> float:
 
     def write_mortality_results(
         self,
-        cause_state: Union[str, DiseaseState, RiskAttributableDisease],
         measure: str,
         results: pd.DataFrame,
     ) -> None:
-        measure_name = measure.split("_due_to_")[0]
-        kwargs = {
-            "entity_type": (
-                cause_state.cause_type if not isinstance(cause_state, str) else "cause"
-            ),
-            "entity": cause_state.model if not isinstance(cause_state, str) else cause_state,
-            "sub_entity": cause_state.state_id if not isinstance(cause_state, str) else None,
-            "results_dir": self.results_dir,
-            "random_seed": self.random_seed,
-            "input_draw": self.input_draw,
-        }
-        write_dataframe_to_parquet(
+        """Format dataframe and write out"""
+
+        results = results.reset_index()
+
+        if "cause_of_death" in results.columns:
```

Comment:
Would it be better to have 
```suggestion
        if not self.aggregate_causes:
```

Reply:
lol...I don't quite know why I did it this way. I even refer to `self.aggregate_causes` in the comment below